src/test/java/duke/DeadlineTest.java

```diff
@@ -0,0 +1,36 @@
+package duke;
+
+import org.junit.jupiter.api.Test;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Test class to test the functionality of the Deadline class.
+ */
+public class DeadlineTest {
+
+    /**
+     * Test the getFormattedDeadline method of the Deadline class.
+     * It checks if the formatted deadline matches the expected result.
+     */
+    @Test
+    public void testGetFormattedDeadline_taskWithDeadline_formattedDateTime() {
```

Comment:
I like the comprehensive test method names which make it clear what the tests cover.